var/spack/repos/builtin/packages/fftw/package.py

```diff
@@ -77,6 +78,60 @@ class Fftw(AutotoolsPackage):
     provides('fftw-api@2', when='@2.1.5')
     provides('fftw-api@3', when='@3:')
 
+    def flag_handler(self, name, flags):
+        spec = self.spec
+        arch = ""
+        simd_x86 = [
+            'sse', 'sse2', 'avx', 'avx2', 'avx512', 'avx-128-fma',
+            'kcvi'
+        ]
+        simd_ppc = ['altivec', 'vsx']
+        simd_arm = ['neon']
+
+        if (spec.satisfies("platform=cray")):
+            # FIXME; It is assumed that cray is x86_64.
+            # If you support arm on cray, you need to fix it.
+            arch  = "x86_64"
+        if (arch != "x86_64" and not spec.satisfies("target=x86_64")):
```

Comment:
The only place where `arch` is set is in the line above (so I don't think `arch` is ever `"ppc64"`). I think that `arch` will mostly be `""`.

I think you can simplify this if statement to:

```
if not spec.satisfies("target=x86_64"):
    ...
```

If desired, you could further simplify this to something like:

```
targets_to_opts = {
  ('x86_64',):  ['sse', 'sse2', 'avx', 'avx2', 'avx512', 'avx-128-fma', 'kcvi'],
  ('ppc64', 'ppc64le'): ['altivec', 'vsx'],
  ...
}

for targets, opts in targets_to_opts.items():
  if not any(spec.satisfies('target={0}'.format(t)) for t in targets):
    if any(spec.satisfies('simd={0}'.format(x)) for x in opts):
      raise ConflictsInSpecError...
```

Reply:
> I think you can simplify this if statement to:
> 
> ```
> if not spec.satisfies("target=x86_64"):
>     ...
> ```

But on Cray, target is determined by Environment variable CRAY_CPU_TARGET:

$spack arch
cray-cnl6-mic_knl

And I don't know all valid target on cray.

> 
> If desired, you could further simplify this to something like:
> 
> ```
> targets_to_opts = {
>   ('x86_64',):  ['sse', 'sse2', 'avx', 'avx2', 'avx512', 'avx-128-fma', 'kcvi'],
>   ('ppc64', 'ppc64le'): ['altivec', 'vsx'],
>   ...
> }
> 
> for targets, opts in targets_to_opts.items():
>   if not any(spec.satisfies('target={0}'.format(t)) for t in targets):
>     if any(spec.satisfies('simd={0}'.format(x)) for x in opts):
>       raise ConflictsInSpecError...
> ```

I rewrite it.
Also I notice neon is supported arm 32 bit only on fftw, so fix it.